src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -115,248 +119,71 @@ public void run()
-                    //There is no need to deal with requests other than HEAD or GET
-                }
-
-
 
                 //if we need to check the against the base case
                 if(work.getMethod().equalsIgnoreCase("GET") && work.getBaseCaseObj().useContentAnalysisMode())
                 {
-                    if(code == 200)
+                    if(code == HttpStatus.SC_OK)
                     {
-                        if(Config.debug)
-                        {
-                            System.out.println("DEBUG Worker[" + threadId + "]: Base Case Check " + url.toString());
-                        }
-
-
-                        //TODO move this option to the Adv options
-                        //if the responce does not match the base case
-                        Pattern regexFindFile = Pattern.compile(".*file not found.*", Pattern.CASE_INSENSITIVE);
-
-                        Matcher m = regexFindFile.matcher(responce);
-
-                        //need to clean the base case of the item we are looking for
-                        String basecase = FilterResponce.removeItemCheckedFor(work.getBaseCaseObj().getBaseCase(), work.getItemToCheck());
-
-                        if(m.find())
-                        {
-                            //do nothing as we have a 404
-                        }
-                        else if(!responce.equalsIgnoreCase(basecase))
-                        {
-                            if(work.isDir())
-                            {
-                                if(Config.debug)
-                                {
-                                    System.out.println("DEBUG Worker[" + threadId + "]: Found Dir (base case)" + url.toString());
-                                }
-                                //we found a dir
-                                manager.foundDir(url, code, responce, basecase, rawResponce, work.getBaseCaseObj());
-                            }
-                            else
-                            {
-                                //found a file
-                                if(Config.debug)
-                                {
-                                    System.out.println("DEBUG Worker[" + threadId + "]: Found File (base case)" + url.toString());
-                                }
-                                manager.foundFile(url, code, responce, work.getBaseCaseObj().getBaseCase(), rawResponce, work.getBaseCaseObj());
-                            }
-                        }
+                        verifyResponseForValidRequests(code, response, rawResponse);
                     }
-                    else if(code == 404 || code == 400)
+                    else if(code == HttpStatus.SC_NOT_FOUND || code == HttpStatus.SC_BAD_REQUEST)
                     {
-                        //again do nothing as it is not there
+                    	System.out.println("DEBUG Worker[" + threadId + "]: " + code + " for: " + url.toString());
```

Comment:
Should check Config.debug before printing to System.out. Note that in the main ZAP code System.out shouldnt be used - we use log4j - be nice to change the dirbuster code to use this in time ;)